go-crud/books/controllers.go

```diff
@@ -0,0 +1,257 @@
+package books
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+)
+
+var data []book
```

Comment:
Keeping this as a package level variable is a code smell. It could cause problems. If you want to keep state I suggest you to create a struct type and store this in there and you can add your handler methods to that type.